File: TelephoneNumber.py
#Mikaela-Rose Porter
"""Write a program that asks the user to enter a 10-character
telephone number in the format XXX-XXX-XXXX. The application should
display the telephone number with any alphabetic characters that
appeared in the original translated to their numeric equivalent.
Algorithm:
1.) Get the number from the user which may contain letters, digits, and -
    create an empty spring to store the result - newNum

2.) Using a for loop
    a) As ve process the characters make them upper case
    b) Use a complicated if/elf/elif
    if letter is in "ABC":
        newNum, append (*2')
    elif in DEF, append 3
    elif in GHI, append 4
    elif in JKL, append 5
    elif in MNO, append 6
    elif in PQRS, append 7
    elif in TUV, append 8
    elif in WXYZ, append 9
    else append the character (for digits and -)

"""
# Prompt the user to enter a phone number
phoneNum = input("Enter a phone number: ")
newNum = ""

for char in phoneNum:
    let = char.upper()
    if let in "ABC":
        newNum.append("2")
    elif let in "DEF":
        newNum.append("3")
    elif let in "GHI":
        newNum.append("4")
    elif let in "JKL":
        newNum.append("5")
    
    elif let in "MNO":
        newNum.append("6")
    elif let in "PQRS":
        newNum.append("7")
    elif let in "TUV":
        newNum.append("8")
    elif let in "WXYZ":
        newNumm.append("9")
    else:
        newNum = newNum + char

print("The old number was ", phoneNum, ". The new number is ", newNum)

File: C++/Classwork/TelephoneNumber.cpp
//Mikaela-Rose Porter
/*Purpose: a program that asks the user to enter a 
10-character phone number as XXX-XXX-XXXX, Then returns the number in all digits.
Input: Integer numbers and Characters. Output: Integers 
Algorithm:
1. Get phone number
2. Initialize a string for the converted number
3. For each letter in the phone number
4. Convert the letter to a number
5. Concatenate the letter with the string for the converted number
6. Display the converted number*/

#include <iostream>
#include <string>
#include <cctype>
using namespace std;

// Function to convert a letter to its corresponding digit
char convertLetter(char letter) {
    // Convert letter to uppercase
    char digit = toupper(letter);

    // Map letters to digits
    if (digit == 'A' || digit == 'B' || digit == 'C')
        return '2';
    else if (digit == 'D' || digit == 'E' || digit == 'F')
        return '3';
    else if (digit == 'G' || digit == 'H' || digit == 'I')
        return '4';
    else if (digit == 'J' || digit == 'K' || digit == 'L')
        return '5';
    else if (digit == 'M' || digit == 'N' || digit == 'O')
        return '6';
    else if (digit == 'P' || digit == 'Q' || digit == 'R' || digit == 'S')
        return '7';
    else if (digit == 'T' || digit == 'U' || digit == 'V')
        return '8';
    else if (digit == 'W' || digit == 'X' || digit == 'Y' || digit == 'Z')
        return '9';
    else
        return letter; // Return unchanged if it's not a letter
}

int main() {
    string phoneNumber;
    string newNumber = "";

    // Prompt user for input
    cout << "Enter the phone number as XXX-XXX-XXXX: ";
    getline(cin, phoneNumber);

    // Convert each character in the phone number
    for (char letter : phoneNumber) {
        newNumber += convertLetter(letter);
    }

    // Display the converted number
    cout << "The phone number is: " << newNumber << endl;

    return 0;
}
